azure-kusto-data/source/models.ts

```diff
@@ -31,21 +30,35 @@ interface Column {
     DateType?: string
 }
 
-
 export class KustoResultRow {
     columns: KustoResultColumn[];
     raw: any;
 
     [column: string]: any;
 
-    constructor(columns: KustoResultColumn[], row: { [ord: number]: any }) {
+    constructor(columns: KustoResultColumn[], row: { [ord: number]: any }, dateTimeParser: DateTimeParser = defaultDatetimeParser, timeSpanParser: TimeSpanParser = defaultTimespanParser) {
         this.columns = columns.sort((a, b) => a.ordinal - b.ordinal);
         this.raw = row;
 
         for (const col of this.columns) {
-            const parse = ValueParser[col.type as string];
+            if (col.name == null) {
+                continue;
+            }
+            let value = row[col.ordinal];
+
+            if (col.type != null) {
```

Comment:
Do you mean a global list and the users will change the value globally?

Or do you mean like before, a list with the default parsers?

Because after indexing into the parser, we would still need to do a type check to assign it the user given parser:
```js
let value = row[col.ordinal];
let parserMethod = ValueParsers[col.type];
if (col.type === "datetime" && dateTimeParser != null) {
   parserMethod = dateTimeParser ;
}
if (col.type === "timespan" && timeSpanParser != null) {
   parserMethod = timeSpanParser ;
}
```

If you have a better way I'll change it, but right now the switch case reduces this duplication